### PR TITLE
Native: entropy device/driver for native_posix

### PR DESCRIPTION
**Pull request by aescolar**
Added pseudo-random entropy device for the native_posix board
+
Changes in samples/drivers/entropy so it is tested in the native_posix regression

### CODE REVIEW

File: samples/drivers/entropy/README.txt [hunk 1]
```diff
@@ -9,7 +9,7 @@ Sample for the entropy gathering driver
 Building and Running Project:
 
 This project writes an infinite series of random numbers to the
-console, 10 per second.
+console, 9 per second.
```
Comment (dbkinder) [inline samples/drivers/entropy/README.txt]: Or is it simply as fast as possible, in which case I’d delete the last part saying “9 per second”

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: samples/drivers/entropy/README.txt (hunk 1 above)
Comment (aescolar-ot) [reply samples/drivers/entropy/README.txt]: It actually loops fetching 9 values, and waiting 1 second.
Seems this commit 3ff4d7409ff44bc7e088d0103cb1138a7f859379 reduced the number of random values to 9 (by using the 10th entry as an overflow marker), but did not change the description

File: drivers/entropy/entropy_native_posix.c [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+#include <stdlib.h>
+
+static unsigned int seed = 0x5678;
+
+void entropy_native_posix_set_seed(unsigned int seed_i)
+{
+	seed = seed_i;
+}
+
+static int entropy_native_posix_get_entropy(struct device *dev, u8_t *buffer,
```
Comment (lpereira) [inline drivers/entropy/entropy_native_posix.c]: Using `memcpy()` should make this slightly smaller (and a tiny wee little bit faster):

```C
while (length) {
        long int value = random();
        size_t to_copy = min(length, sizeof(value));

        memcpy(buffer, &value, to_copy);
        buffer += to_copy;
        len -= to_copy;
}
```

**Review by lpereira — COMMENTED** [review]
Do we have a need to set the seed? Because if not, I would really prefer to just read from `/dev/urandom` -- it's available everywhere, won't block, and, at least on Linux, is pretty decent.

**Comment by aescolar-ot** [comment]
>Do we have a need to set the seed? 

Absolutely. We need to be able to reproduce tests. We don't want unpredictability.
+
We don't really need a "good" random number generator. Just something good enough to exercise the code with different data. For that `random()` should be way more than enough. We don't run in native_posix to generate cryptographically sound data, just to test.

File: drivers/entropy/entropy_native_posix.c (hunk 2 above)
Comment (aescolar-ot) [reply drivers/entropy/entropy_native_posix.c]: Actually when I wrote that, I was leaning to just take 1 byte from each `random()` call for the shake of simplicity. The amount of time spent generating random bytes being probably quite minimal anyhow.

**Review by dbkinder — APPROVED** [review]

File: samples/drivers/entropy/README.txt (hunk 1 above)
Comment (dbkinder) [reply samples/drivers/entropy/README.txt]: OK. +1 for docs then

**Review by nashif — APPROVED** [review]

**Comment by d3zd3z** [comment]
I'm a little concerned about not having entropy in the native posix environment. There is a concern about repeatability, which is somewhat valid, but if this code is _ever_ used to make real connections to external services, those connections will be completely insecure, possibly revealing more than is expected.

The behavior of cryptographic operations should not depend on a particular sequence coming from a random source (in fact, if it does, this can be considered a security vulnerability, as it suggests information can be revealed through timing attacks). Do we know of actual tests that produce different results, or is this just speculation?

**Pull request merged**